src/System.Net.Http/tests/FunctionalTests/HttpClientTestBase.cs

```diff
@@ -59,5 +65,19 @@ protected static object GetUnderlyingSocketsHttpHandler(HttpClientHandler handle
             FieldInfo field = typeof(HttpClientHandler).GetField("_socketsHttpHandler", BindingFlags.Instance | BindingFlags.NonPublic);
             return field?.GetValue(handler);
         }
+
+        // Common [MemberData] for [Theory] tests
+
+        public static readonly LoopbackServerFactory[] LoopbackServerFactories = new LoopbackServerFactory[]
+        {
+            Http11LoopbackServerFactory.Singleton,
+            Http2LoopbackServerFactory.Singleton
+        };
+
+        public static readonly object[][] LoopbackServerFactoriesMemberData = new object[][] 
+        {
+            new object[] { Http11LoopbackServerFactory.Singleton },
+            new object[] { Http2LoopbackServerFactory.Singleton }
+        };
```

Comment:
Nit: just to avoid the duplication, this could be:
```C#
public static readonly object[][] LoopbackServerFactoriesMemberData =
    LoopbackServerFactories.Select(f => new object[] { f }).ToArray();
```
or something like that.